migrations/V188__Add_MTO_Category_Reorder_Trigger.sql

```diff
@@ -29,17 +29,50 @@ BEGIN
     
     -- Handle Update: Reorder positions when a row's position is changed 
     ELSIF TG_OP = 'UPDATE' THEN
-        -- Determine if record moved up or down
-        IF NEW.position > OLD.position THEN
-            position_adjustment := -1;
-        ELSE
-            position_adjustment := 1;
-        END IF;
+    IF OLD.model_plan_id IS DISTINCT FROM NEW.model_plan_id THEN
+        RAISE EXCEPTION 'updating model_plan_id is not allowed. Caught in trigger function update_position_based_on_parent_and_model_plan';
+    END IF;
 
  /* Prevent updates to model_plan_id and parent_id as these are fixed for each category */
```

Comment:
Should probably just move this comment up and remove the bit about parent_id
```suggestion
    /* Prevent updates to model_plan_id as categories cannot move from one model plan to another */
    IF OLD.model_plan_id IS DISTINCT FROM NEW.model_plan_id THEN
        RAISE EXCEPTION 'updating model_plan_id is not allowed. Caught in trigger function update_position_based_on_parent_and_model_plan';
    END IF;

```